.github/workflows/ci.yml

```diff
@@ -0,0 +1,197 @@
+                             
+        - name: Source-based Dependencies [Windows]
+          if: matrix.os == 'windows-latest'
+          shell: bash
+          run: |
+            # YCM
+            cd ${GITHUB_WORKSPACE}
+            git clone https://github.com/robotology/ycm
+            cd ycm
+            mkdir -p build
+            cd build
+            cmake -A x64 -DCMAKE_TOOLCHAIN_FILE=${VCPKG_INSTALLATION_ROOT}/scripts/buildsystems/vcpkg.cmake \
+                         -DCMAKE_INSTALL_PREFIX=${GITHUB_WORKSPACE}/install ..
+            cmake --build . --config ${{matrix.build_type}} --target INSTALL
+             
+            # YARP
+            cd ${GITHUB_WORKSPACE}
+            git clone https://github.com/robotology/yarp
+            cd yarp
+            mkdir -p build
+            cd build
+            cmake -A x64 -DCMAKE_TOOLCHAIN_FILE=${VCPKG_INSTALLATION_ROOT}/scripts/buildsystems/vcpkg.cmake \
+                         -DCMAKE_PREFIX_PATH=${GITHUB_WORKSPACE}/install \
+                         -DCMAKE_INSTALL_PREFIX=${GITHUB_WORKSPACE}/install ..
+            cmake --build . --config ${{matrix.build_type}} --target INSTALL
+            # Workaround for https://github.com/robotology-dependencies/robotology-vcpkg-binary-ports/issues/3
+            export IPOPT_DIR=${VCPKG_INSTALLATION_ROOT}/installed/x64-windows
+                          
+            # ICUB
+            cd ${GITHUB_WORKSPACE}
+            git clone https://github.com/robotology/icub-main
+            cd icub-main
+            mkdir -p build
+            cd build
+            cmake -A x64 -DCMAKE_TOOLCHAIN_FILE=${VCPKG_INSTALLATION_ROOT}/scripts/buildsystems/vcpkg.cmake \
+                         -DCMAKE_PREFIX_PATH=${GITHUB_WORKSPACE}/install \
+                         -DCMAKE_BUILD_TPYE=${{matrix.build_type}} \
+                         -DCMAKE_INSTALL_PREFIX=${GITHUB_WORKSPACE}/install ..
+            cmake --build . --config ${{matrix.build_type}} --target INSTALL
+            
+            # iDynTree
+            cd ${GITHUB_WORKSPACE}
+            git clone https://github.com/robotology/iDynTree
+            cd iDynTree
+            git checkout devel
+            mkdir -p build
+            cd build
+            cmake -A x64 -DCMAKE_TOOLCHAIN_FILE=${VCPKG_INSTALLATION_ROOT}/scripts/buildsystems/vcpkg.cmake \
+                         -DCMAKE_PREFIX_PATH=${GITHUB_WORKSPACE}/install \
+                         -DCMAKE_BUILD_TPYE=${{matrix.build_type}} \
+                         -DIDYNTREE_USES_ICUB_MAIN:BOOL=ON \
+                         -DCMAKE_INSTALL_PREFIX=${GITHUB_WORKSPACE}/install ..
+            cmake --build . --config ${{matrix.build_type}} --target INSTALL
+                           
+             
+        - name: Source-based Dependencies [Ubuntu/macOS]
+          if: matrix.os == 'ubuntu-latest' || matrix.os == 'macOS-latest'
+          shell: bash
+          run: |
+            # YCM
```

Comment:
Just to know, is `YCM` required to build `whole-body-estimators`? If yes maybe we add it to the README?

Reply:
YCM is actually needed for handling some CMake features for YARP and related dependencies.

Reply:
I'll update the README

Reply:
Done in https://github.com/robotology/whole-body-estimators/pull/47/commits/268de81886c387a609e16d11ca40654f6ea6a3de